test/sql/simple/test_ilike.cpp

```diff
@@ -0,0 +1,79 @@
+#include "catch.hpp"
+#include "test_helpers.hpp"
+
+using namespace duckdb;
+using namespace std;
+
+TEST_CASE("Test scalar ILIKE statement", "[ilike]") {
+	unique_ptr<QueryResult> result;
+	DuckDB db(nullptr);
+	Connection con(db);
+
+	// scalar ILIKE
+	result = con.Query("SELECT 'aaa' ILIKE 'bbb'");
+	REQUIRE(CHECK_COLUMN(result, 0, {Value::BOOLEAN(false)}));
+
+	result = con.Query("SELECT 'aaa' ILIKE 'bBb'");
+	REQUIRE(CHECK_COLUMN(result, 0, {Value::BOOLEAN(false)}));
+
+	result = con.Query("SELECT 'aaa' ILIKE 'aaa'");
+	REQUIRE(CHECK_COLUMN(result, 0, {Value::BOOLEAN(true)}));
+
+	result = con.Query("SELECT 'aAa' ILIKE 'aaa'");
+	REQUIRE(CHECK_COLUMN(result, 0, {Value::BOOLEAN(true)}));
+
+	result = con.Query("SELECT 'aaa' ILIKE 'aAa'");
+	REQUIRE(CHECK_COLUMN(result, 0, {Value::BOOLEAN(true)}));
+
+	result = con.Query("SELECT 'aaa' ILIKE '%'");
+	REQUIRE(CHECK_COLUMN(result, 0, {Value::BOOLEAN(true)}));
+
+	result = con.Query("SELECT 'aaa' ILIKE '%A'");
+	REQUIRE(CHECK_COLUMN(result, 0, {Value::BOOLEAN(true)}));
+
+	result = con.Query("SELECT 'aaa' ILIKE '%b'");
+	REQUIRE(CHECK_COLUMN(result, 0, {Value::BOOLEAN(false)}));
+
+	result = con.Query("SELECT 'aaa' ILIKE 'a%'");
+	REQUIRE(CHECK_COLUMN(result, 0, {Value::BOOLEAN(true)}));
+
+	result = con.Query("SELECT 'aaa' ILIKE 'b%'");
+	REQUIRE(CHECK_COLUMN(result, 0, {Value::BOOLEAN(false)}));
+
+	result = con.Query("SELECT 'aaa' ILIKE 'a_A'");
+	REQUIRE(CHECK_COLUMN(result, 0, {Value::BOOLEAN(true)}));
+
+	result = con.Query("SELECT 'aaa' ILIKE 'a_'");
+	REQUIRE(CHECK_COLUMN(result, 0, {Value::BOOLEAN(false)}));
+
+	result = con.Query("SELECT 'aaa' ILIKE '__%'");
+	REQUIRE(CHECK_COLUMN(result, 0, {Value::BOOLEAN(true)}));
+
+	result = con.Query("SELECT 'aaa' ILIKE '____%'");
+	REQUIRE(CHECK_COLUMN(result, 0, {Value::BOOLEAN(false)}));
+
+	result = con.Query("SELECT 'aBaBaC' ILIKE '%abac'");
+	REQUIRE(CHECK_COLUMN(result, 0, {Value::BOOLEAN(true)}));
+
+	result = con.Query("SELECT 'aBaBaC' NOT ILIKE '%abac'");
```

Comment:
Please also add some more tests involving non-ascii characters (e.g. `SELECT 'Á' ILIKE 'á'; -- true`) and tests involving NULLs.